analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -44,6 +45,37 @@ protected override ProgramState LearnBranchingConstraint(ProgramState state, IBi
         return state;
     }
 
+    protected override ProgramState PreProcess(ProgramState state, IBinaryOperationWrapper operation)
+    {
+        if (Calculation(operation.OperatorKind) is { } calculation
+            && operation.LeftOperand?.TrackedSymbol() is { } leftSymbol
+            && operation.RightOperand?.TrackedSymbol() is { } rightSymbol
+            && state[leftSymbol]?.Constraint<NumberConstraint>() is { } leftNumber
+            && state[rightSymbol]?.Constraint<NumberConstraint>() is { } rightNumber)
+        {
+            state = state.SetOperationConstraint(operation.WrappedOperation, Calculate(leftNumber, rightNumber, calculation));
```

Comment:
Let's avoid the lambdas for memory reasons, because we can.
First extract left and right, then send it to other method to return a new constraint. If not null, update the operation.

That should simplify it here, make it suitable for ternary, and therefore arrow method. 

Similar approach is also in the methods below.